nav2_costmap_2d/src/layered_costmap.cpp

```diff
@@ -274,10 +274,13 @@ bool LayeredCostmap::isCurrent()
 
 void LayeredCostmap::setFootprint(const std::vector<geometry_msgs::msg::Point> & footprint_spec)
 {
-  footprint_ = footprint_spec;
-  nav2_costmap_2d::calculateMinAndMaxDistances(
-    footprint_spec,
-    inscribed_radius_, circumscribed_radius_);
+  std::pair<double, double> inside_outside = nav2_costmap_2d::calculateMinAndMaxDistances(
+    footprint_spec);
+  std::atomic_store(
+    &footprint_,
+    std::make_shared<std::vector<geometry_msgs::msg::Point>>(footprint_spec));
```

Comment:
This can be a unique pointer if you're not returning the pointer to anyone else but always dereferencing it to send a reference of the object in any `getFootprint` calls. Shared pointer has overhead, unique has effectively none.

Also, add a note here that we have these as atomics because the sets here and gets in algorithms' updates aren't locked on the same thread 

After that, this LGTM

Reply:
Also, a couple of unusual tests failed on this PR that aren't considered flaky. I'm retriggering, but if that happens again, you may need to look into issues this is introducing

Reply:
`atomic_load` and `atomic_store` do not support `std::unique_ptr`.
I think keeping the shared_ptr would be the best option.